File: algorithms/binary_search/binary_search.py
# BINARY SEARCH IN PYTHON

# algorithms/binary_search/binary_seach.py

def binary_search(arr, x):
    low = 0
    high = len(arr) - 1
    mid = 0

    while low <= high:
        mid = (high + low) // 2

        # If x is greater, ignore left half
        if arr[mid] < x:
            low = mid + 1

        # If x is smaller, ignore right half
        elif arr[mid] > x:
            high = mid - 1

        # x is present at mid
        else:
            return mid

    # If we reach here, then the element was not present
    return -1

File: algorithms/binary_search/c/binary_search.c
// algorithms/binary_search/c/binary_search.c

#include "binary_search.h"

int binary_search(int arr[], int size, int x) {
    int low = 0;
    int high = size - 1;
    int mid;

    while (low <= high) {
        mid = (high + low) / 2;

        if (arr[mid] < x) {
            low = mid + 1;
        } else if (arr[mid] > x) {
            high = mid - 1;
        } else {
            return mid;  // Element found at mid
        }
    }

    return -1;  // Element not found
}
